charts/ceph-csi-cephfs/templates/storageclass.yaml

```diff
@@ -0,0 +1,28 @@
+{{- if .Values.storageClass.create -}}
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+   name: {{ .Values.storageClass.name }}
+   namespace: {{ .Release.Namespace }}
+   labels:
+     app: {{ include "ceph-csi-cephfs.name" . }}
+     chart: {{ include "ceph-csi-cephfs.chart" . }}
+     release: {{ .Release.Name }}
+     heritage: {{ .Release.Service }}
+provisioner: {{ .Values.driverName }}
+parameters:
+  clusterID: {{ .Values.storageClass.parameters.clusterID }}
+  fsName: {{ .Values.storageClass.parameters.fsName }}
+  csi.storage.k8s.io/provisioner-secret-name: {{ .Values.secret.metadata.name }}
+  csi.storage.k8s.io/provisioner-secret-namespace: {{ .Release.Namespace }}
+  csi.storage.k8s.io/controller-expand-secret-name: {{ .Values.secret.metadata.name }}
+  csi.storage.k8s.io/controller-expand-secret-namespace: {{ .Release.Namespace }}
+  csi.storage.k8s.io/node-stage-secret-name: {{ .Values.secret.metadata.name }}
+  csi.storage.k8s.io/node-stage-secret-namespace: {{ .Release.Namespace }}
```

Comment:
what if someone wants to use a different secretes for different opetaions?

Reply:
Users can now mention different secret names in the values.yaml.

Reply:
Yes, but we are restricting the namespace where he wants to create secrets, and looks like no option to specify the namespace name also.

Reply:
@Yuggupta27 is this fixed?

Reply:
Thanks for the Suggestion @Madhu-1, Now added the support for storageclass to make use of secrets created in different namespaces.

When no namespace is passed and charts are deployed in a `testing` namespace
```
Name:                  csi-rbd-sc1
IsDefaultClass:        No
Annotations:           meta.helm.sh/release-name=ceph-csi-rbd,meta.helm.sh/release-namespace=testing
Provisioner:           rbd.csi.ceph.com
Parameters:            clusterID=<cluster-ID>,csi.storage.k8s.io/controller-expand-secret-name=csi-rbd-secret,csi.storage.k8s.io/controller-expand-secret-namespace=testing,csi.storage.k8s.io/fstype=ext4,csi.storage.k8s.io/node-stage-secret-name=csi-rbd-secret,csi.storage.k8s.io/node-stage-secret-namespace=testing,csi.storage.k8s.io/provisioner-secret-name=csi-rbd-secret,csi.storage.k8s.io/provisioner-secret-namespace=testing,imageFeatures=layering,pool=<pool-name>,thickProvision=false
AllowVolumeExpansion:  True
MountOptions:          <none>
ReclaimPolicy:         Delete
VolumeBindingMode:     Immediate
Events:                <none>
```


When Secrets are present in separate namespaces (say `test1`, `test2` and `test3`)
```
Name:                  csi-rbd-sc1
IsDefaultClass:        No
Annotations:           meta.helm.sh/release-name=ceph-csi-rbd,meta.helm.sh/release-namespace=testing
Provisioner:           rbd.csi.ceph.com
Parameters:            clusterID=<cluster-ID>,csi.storage.k8s.io/controller-expand-secret-name=csi-rbd-secret,csi.storage.k8s.io/controller-expand-secret-namespace=test2,csi.storage.k8s.io/fstype=ext4,csi.storage.k8s.io/node-stage-secret-name=csi-rbd-secret,csi.storage.k8s.io/node-stage-secret-namespace=test3,csi.storage.k8s.io/provisioner-secret-name=csi-rbd-secret,csi.storage.k8s.io/provisioner-secret-namespace=test1,imageFeatures=layering,pool=<pool-name>,thickProvision=false
AllowVolumeExpansion:  True
MountOptions:          <none>
ReclaimPolicy:         Delete
VolumeBindingMode:     Immediate
Events:                <none>
```